common/http/class.Request.php

```diff
@@ -237,15 +242,31 @@ public function send()
         //curl_setopt($curlHandler, CURLINFO_HEADER_OUT, 1);
         //curl_setopt($curlHandler, CURLOPT_HEADER, true);
 
+        //directly setting `FOLLOWLOCATION` to false to make sure next lines would be working as expected
+        //and we can get the redirect url from curl
+        curl_setopt($curlHandler, CURLOPT_FOLLOWLOCATION, 0);
+
         $responseData = curl_exec($curlHandler);
         $httpResponse = new common_http_Response();
 
         $httpResponse->httpCode = curl_getinfo($curlHandler, CURLINFO_HTTP_CODE);
         $httpResponse->headerOut = curl_getinfo($curlHandler, CURLINFO_HEADER_OUT);
         $httpResponse->effectiveUrl = curl_getinfo($curlHandler, CURLINFO_EFFECTIVE_URL);
         $httpResponse->responseData = $responseData;
+
+        $fullInfo = curl_getinfo($curlHandler);
```

Comment:
Just a suggestion. You can add `CURLINFO_REDIRECT_URL` option here, and get the redirect URL value if any straight away. It will simplify the nested conditional statements below.